src/SiteBuilder.php

```diff
@@ -55,15 +75,46 @@ private function cleanup()
         $this->files->deleteDirectory($this->cachePath);
     }
 
-    private function writeFiles($source, $destination, $siteData)
+    private function generateFiles($source, $siteData)
     {
-        return collect($this->files->allFiles($source))->map(function ($file) use ($source) {
+        $this->consoleOutput->writeln('<comment>Generating files from source</comment>');
+
+        $files = collect($this->files->allFiles($source));
+
+        $progressBar = new ProgressBar($this->consoleOutput, $files->count());
+        $progressBar->start();
+
+        $files = $files->map(function ($file) use ($source) {
             return new InputFile($file, $source);
-        })->flatMap(function ($file) use ($siteData) {
+        })->flatMap(function ($file) use ($siteData, $progressBar) {
+            $progressBar->advance();
+
             return $this->handle($file, $siteData);
-        })->map(function ($file) use ($destination) {
+        });
+
+        $progressBar->finish();
+        $this->consoleOutput->writeln('');
+
+        return $files;
+    }
+
+    private function writeFiles($destination, $files)
+    {
+        $this->consoleOutput->writeln('<comment>Writing files to destination</comment>');
+
+        $progressBar = new ProgressBar($this->consoleOutput, $files->count());
+        $progressBar->start();
+
+        $files = $files->map(function ($file) use ($destination, $progressBar) {
+            $progressBar->advance();
+
             return $this->writeFile($file, $destination);
         });
+
+        $progressBar->finish();
+        $this->consoleOutput->writeln('');
+
+        return $files;
```

Comment:
In my testing, this second progress bar isn't all that useful, since writing happens very quickly. I would remove it.